auth0_flutter_platform_interface/test/user_profile_test.dart

```diff
@@ -0,0 +1,137 @@
+import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('UserProfile', () {
+    test('fromMap correctly maps all properties', () {
+      final map = {
+        'sub': 'user123',
+        'name': 'John Doe',
+        'given_name': 'John',
+        'family_name': 'Doe',
+        'middle_name': 'Smith',
+        'nickname': 'Johnny',
+        'preferred_username': 'john.doe',
+        'profile': 'https://example.com/profile',
+        'picture': 'https://example.com/picture.jpg',
+        'website': 'https://johndoe.com',
+        'email': 'john@example.com',
+        'email_verified': true,
+        'gender': 'male',
+        'birthdate': '1990-01-01',
+        'zoneinfo': 'America/New_York',
+        'locale': 'en-US',
+        'phone_number': '+1234567890',
+        'phone_number_verified': true,
+        'address': {'country': 'USA', 'postal_code': '12345'},
+        'updated_at': '2023-01-01T00:00:00.000Z',
+        'custom_claims': {'role': 'admin'}
+      };
+
+      final userProfile = UserProfile.fromMap(map);
+
+      expect(userProfile.sub, 'user123');
+      expect(userProfile.name, 'John Doe');
+      expect(userProfile.givenName, 'John');
+      expect(userProfile.familyName, 'Doe');
+      expect(userProfile.middleName, 'Smith');
+      expect(userProfile.nickname, 'Johnny');
+      expect(userProfile.preferredUsername, 'john.doe');
+      expect(userProfile.profileUrl, Uri.parse('https://example.com/profile'));
+      expect(
+          userProfile.pictureUrl, Uri.parse('https://example.com/picture.jpg'));
+      expect(userProfile.websiteUrl, Uri.parse('https://johndoe.com'));
+      expect(userProfile.email, 'john@example.com');
+      expect(userProfile.isEmailVerified, true);
+      expect(userProfile.gender, 'male');
+      expect(userProfile.birthdate, '1990-01-01');
+      expect(userProfile.zoneinfo, 'America/New_York');
+      expect(userProfile.locale, 'en-US');
+      expect(userProfile.phoneNumber, '+1234567890');
+      expect(userProfile.isPhoneNumberVerified, true);
+      expect(userProfile.address, {'country': 'USA', 'postal_code': '12345'});
+      expect(userProfile.updatedAt, DateTime.parse('2023-01-01T00:00:00.000Z'));
+      expect(userProfile.customClaims, {'role': 'admin'});
+    });
+
+    test('fromMap handles missing optional properties', () {
+      final map = {
+        'sub': 'user123',
+        'email': 'john@example.com',
+      };
```

Comment:
```suggestion
      final map = {'sub': 'user123'};
```
`email` is not required.